ui-tests/tests/search.spec.ts

```diff
@@ -172,3 +172,26 @@ test('should make a new request on refresh', async ({ page }) => {
     page.locator('[title="button to refresh and reload results"]').click()
   ]);
 });
+
+
+test('should expand and collapse tree view on clicking expand-collapse button', async ({ page }) => {
+  // Click #tab-key-0 .lm-TabBar-tabIcon svg >> nth=0
+  await page.locator('[title="Search and replace"]').click();
+  // Fill input[type="search"]
+  await page.locator('input[type="search"]').fill('strange');
+
+  await Promise.all([
+    page.waitForResponse(
+      response =>
+        /.*search\/\?query=strange/.test(response.url()) &&
+        response.request().method() === 'GET'
+    ),
+    page.locator('input[type="search"]').press('Enter'),
+    page.locator('[title="button to expand and collapse all results"]').click()
+  ]);
+
```

Comment:
You are doing too much action at once.

You should first start the query in `Promise.all`. Probably waiting for 

```ts
  await page.waitForSelector('.jp-search-replace-tab >> .jp-progress', {
    state: 'hidden'
  })
```

After it to ensure you get the results displayed.

Then only click on the expand/collapse button.

Currently you are doing something no human can do: validating the input box and clicking on the button at the same time. (The click actually has no effect as there are no results.)

Reply:
fixed